kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/jms/ReqReplyResponseHandler.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.server.client.jms;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.jms.Connection;
+import javax.jms.JMSException;
+import javax.jms.Message;
+import javax.jms.MessageConsumer;
+import javax.jms.Queue;
+import javax.jms.Session;
+import javax.jms.TextMessage;
+
+import org.kie.server.api.jms.JMSConstants;
+import org.kie.server.api.marshalling.Marshaller;
+import org.kie.server.api.model.ServiceResponse;
+import org.kie.server.api.model.ServiceResponsesList;
+import org.kie.server.client.KieServicesClient;
+import org.kie.server.client.KieServicesConfiguration;
+import org.kie.server.client.KieServicesException;
+import org.kie.server.client.impl.KieServicesClientImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ReqReplyResponseHandler implements ResponseHandler {
```

Comment:
This is just a detail but why is this called `ReqReplyResponseHandler` and not `RequestReplyResponseHandler`? It would still be shorter than `FireAndForgetResponseHandler`. The current name might be a little bit ambiguous.